packages/main/src/TimePicker.ts

```diff
@@ -363,56 +371,21 @@ class TimePicker extends UI5Element implements IFormInputElement {
 		this.tempValue = e.detail.value; // every time the user changes the time selection -> update tempValue
 	}
 
-	/**
-	 * Opens the picker.
-	 * @public
-	 * @returns Resolves when the picker is open
-	 */
-	openPicker(): void {
-		this.tempValue = this.value && this.isValid(this.value) ? this.value : this.getFormat().format(UI5Date.getInstance());
-		const responsivePopover = this._getPopover();
-		responsivePopover.opener = this;
-		responsivePopover.open = true;
-	}
-
-	/**
-	 * Closes the picker
-	 * @public
-	 * @returns Resolves when the picker is closed
-	 */
-	closePicker(): void {
-		const responsivePopover = this._getPopover();
-		responsivePopover.open = false;
-		this._isPickerOpen = false;
-	}
-
-	togglePicker() {
-		if (this.isOpen()) {
-			this.closePicker();
-		} else if (this._canOpenPicker()) {
-			this.openPicker();
-		}
-	}
-
-	/**
-	 * Checks if the picker is open
-	 * @public
-	 */
-	isOpen(): boolean {
-		return !!this._isPickerOpen;
+	_togglePicker() {
+		this.open = !this.open;
 	}
 
 	submitPickers() {
 		this._updateValueAndFireEvents(this.tempValue!, true, ["change", "value-changed"]);
-		this.closePicker();
+		this._togglePicker();
 	}
 
 	onResponsivePopoverAfterClose() {
-		this._isPickerOpen = false;
+		this.open = false;
 	}
 
 	onResponsivePopoverAfterOpen() {
-		this._isPickerOpen = true;
+		this.open = true;
```

Comment:
Is this necessary?

Reply:
After a discussion with Stoyan, we agreed that setting an attribute/property directly to the popover isn't a valid use case, as we are the one creating and managing the popover, that's why we don't need to 'care' if someone else changes the state. Also the logic on timepicker side should only be: set open to false -> close, set open to true -> open. So both handlers will be deleted, from TS and hbs files.